### ISSUE TITLE
Add BeforeClass/AfterClass like on ITestListener

### ISSUE THREAD

**Issue by vguna** [cbeust/testng#776]
It would be great, if the ITestListener would offer a method that is invoked, when a Test Class is created and gets executed.

I'm trying to write a TestNG listener that is fired once on each test class to print the test class name to the log/console. I tried it with ITestListener onTestStart(), but this is called per test - as the name states. To log only once, I would have to synchronize it somehow when e.g. running in a multithreaded test (threadPoolSize=..) - otherwise it would log n times (once per thread).

Referencing this discussion:

http://testng.1065351.n5.nabble.com/Getting-an-event-when-test-class-is-created-td21715.html#a21735

**Comment by juherr** [comment]
You are asking a listener way to do @BeforeClass and @AfterClass, isn't it?

**Comment by vguna** [comment]
Exactly. Without the need to touch my test classes.

**Comment by juherr** [comment]
Would you like to try to add it yourself and propose it?

**Comment by vguna** [comment]
I'll check with my manager :).

**Comment by juherr** [comment]
Duplicate of #356
